### ISSUE TITLE
`nonminimal_bool` triggers on `is_some()` through macros

### ISSUE THREAD

**Issue by Kyuuhachi** [rust-lang/rust-clippy#10523]
### Summary

It seems #9457 was supposed to fix this, but it does not seem to have been fully successful.

### Lint Name

nonminimal_bool

### Reproducer

I tried this code:

```rust
macro_rules! a {
	($v:expr) => { $v.is_some() }
}

fn main() {
	let x: Option<u32> = None;
	if !a!(x) { }
}
```

I saw this happen:

```
warning: this boolean expression can be simplified
 --> src/lib.rs:7:5
  |
7 |     if !a!(x) { }
  |        ^^^^^^ help: try: `x.is_none()`
  |
```

I expected to see this happen:

### Version

```text
rustc 1.69.0-nightly (7aa413d59 2023-02-19)
binary: rustc
commit-hash: 7aa413d59206fd511137728df3d9e0fd377429bd
commit-date: 2023-02-19
host: x86_64-unknown-linux-gnu
release: 1.69.0-nightly
LLVM version: 15.0.7
```

**Comment by dswij** [comment]
FWIW,https://github.com/rust-lang/rust-clippy/pull/9457 was to fix the suggestion picking up macro definition instead of the callsite. Though, this issue is valid regardless :)
